dashboard/app/helpers/aichat_helper.rb

```diff
@@ -0,0 +1,64 @@
+module AichatHelper
+  ASSISTANT = "assistant"
+  USER = "user"
+  BASE_PROMPT = "You are a helpful chatbot for children. Limit your responses to a small paragraph. "
+  INSTRUCTIONS_BEGIN_TOKEN = "[INST]"
+  INSTRUCTIONS_END_TOKEN = "[/INST]"
+  SENTENCE_BEGIN_TOKEN = "<s>"
+  SENTENCE_END_TOKEN = "</s>"
+  MAX_NEW_TOKENS = 300
+  SAGEMAKER_CLIENT = Aws::SageMakerRuntime::Client.new
+  SAGEMAKER_MODEL_ENDPOINT = "mistral-7b-inst-v01" # "BioMistral-7B"
+  TOP_P = 0.9
+
+  # The instruction-tuned version of Mistral accepts formatted instructions where conversation roles
+  # must start with a user prompt and alternate between user and assistant.
+  # Mistral-7B-Instruction LLM instruction format doc at https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.1.
+  def self.format_inputs_for_sagemaker_request(aichat_params, stored_messages, new_message)
+    all_messages = [*stored_messages, {role: USER, content: new_message}]
+    inputs = BASE_PROMPT + aichat_params[:systemPrompt]
+    inputs += aichat_params[:retrievalContexts].join(" ") if aichat_params[:retrievalContexts]
+    inputs = SENTENCE_BEGIN_TOKEN + wrap_as_instructions(inputs)
+    # Filter only messages from user or assistant - ignore model update messages.
```

Comment:
We should ideally filter out model update messages on the front end before sending

Reply:
This change should result in model update messages being filtered before sending:

https://github.com/code-dot-org/code-dot-org/pull/58103/files